OpenRA.Mods.Common/Traits/LinkOnRepairCursor.cs

```diff
@@ -0,0 +1,49 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Send actor to link to repair cursor.")]
+	public class LinkOnRepairCursorInfo : TraitInfo, Requires<ILinkClientManagerInfo>
+	{
+		[Desc("Linking type")]
+		public readonly BitSet<LinkType> Type = new("Repair");
+
+		public override object Create(ActorInitializer init) { return new LinkOnRepairCursor(init.Self, this); }
+	}
+
+	public class LinkOnRepairCursor
```

Comment:
The abstraction here feels off to me. This should really be part of an OrderTargeter?

Reply:
Because the modder may want to specify which actors this should work on. This abstraction also unties it repair cursor from Repairable trait https://github.com/OpenRA/OpenRA/pull/20665#discussion_r1450788616